src/node_metadata.cc

```diff
@@ -52,14 +52,19 @@ Metadata metadata;
 
 #if HAVE_OPENSSL
 static constexpr size_t search(const char* s, char c, size_t n = 0) {
-  return *s == c ? n : search(s + 1, c, n + 1);
+  return *s == '\0' ? n : (*s == c ? n : search(s + 1, c, n + 1));
```

Comment:
We can just have a test for the output of this function

Reply:
@anonrig i'm assuming i'd put this in `test/cctest` somewhere - what would you prefer?

Reply:
Since the path is only going to be hit by embedders who embed to an implementation that has a version string that doesn’t match the pattern, I am fine without adding a test (adding just a test for this one function doesn’t seem very meaningful since it’s just one piece of the entire behavior, this function being correct doesn’t mean the version string can be correctly coerced to 0.0.0 anyway).